src/test/java/com/lgtm/easymoney/controllers/GroupControllerTest.java

```diff
@@ -1,20 +1,268 @@
+import com.lgtm.easymoney.payload.LeaveGroupReq;
+import com.lgtm.easymoney.payload.ResourceCreatedRsp;
+import com.lgtm.easymoney.services.GroupService;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.ResultActions;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@RunWith(SpringRunner.class)
+@WebMvcTest(GroupController.class)
+public class GroupControllerTest {
+  @Autowired
+  private MockMvc mvc;
+
+  @MockBean
+  private GroupService groupService;
+
+  private CreateGroupReq createGroupReq;
+
+  private InviteToGroupReq inviteToGroupReq;
+
+  private LeaveGroupReq leaveGroupReq;
+
+  private ResourceCreatedRsp createdRsp;
+
+  private GroupRsp groupRsp;
+
+  private String groupName = "test";
+
+  private String groupDescription = "test";
+
+  private Long expectedGid = 1L;
+
+  private Long uid1 = 1L;
+
+  private List<Long> uids = List.of(uid1);
+
+  private Long uid2 = 2L;
+
+  @Before
+  public void setUp() {
+    createGroupReq = new CreateGroupReq();
+    createGroupReq.setName(groupName);
+    createGroupReq.setDescription(groupDescription);
+    createGroupReq.setUids(uids);
+
+    createdRsp = new ResourceCreatedRsp(expectedGid);
+
+    inviteToGroupReq = new InviteToGroupReq();
+    inviteToGroupReq.setGid(expectedGid);
+    inviteToGroupReq.setInviterId(uid1);
+    inviteToGroupReq.setInviteeId(uid2);
+
+    leaveGroupReq = new LeaveGroupReq();
+    leaveGroupReq.setGid(expectedGid);
+    leaveGroupReq.setUid(uid1);
+
+    groupRsp = new GroupRsp();
+    groupRsp.setGid(expectedGid);
+    groupRsp.setName(groupName);
+    groupRsp.setDescription(groupDescription);
+    groupRsp.setUids(uids);
+  }
+
+  @Test
+  public void createGroupSuccess() throws Exception {
+    Mockito.when(groupService.createGroup(createGroupReq))
+        .thenReturn(createdRsp);
+
+    var resultActions = postCreate(createGroupReq);
+
+    resultActions.andExpectAll(
+        status().isCreated(),
+        jsonPath("$.id").value(expectedGid));
+  }
+
+  @Test
+  public void createGroupFailedByEmptyUids() throws Exception {
```

Comment:
qq: what tests should we do for `controller`? what tests for `service`?
for errors caught and threw at `service` level, do we still need testing it again at controller?

@howieraem @jenniferduan45